vllm/v1/core/block_pool.py

```diff
@@ -19,6 +19,101 @@
 logger = init_logger(__name__)
 
 
+class BlockHashToBlockMap:
+    """
+    Cache of blocks that are used for prefix caching.
+    """
+
+    def __init__(self):
+        # Caches blocks from hash directly to a block or multiple blocks
+        # (i.e. {block_hash: KVCacheBlocks})
+        # - Mostly block_hash maps to a single KVCacheBlock, and KVCacheBlocks
+        #   would simply be a KVCacheBlock.
+        # - Otherwise, KVCacheBlocks is a dict from {block_id: KVCacheBlock}
+        #
+        # A cached block is a full block with a block hash that can be used
+        # for prefix caching.
+        # The cached block may be used by running requests or in the
+        # free_block_queue that could potentially be evicted.
```

Comment:
Would be better to move this to the class docstring

Reply:
+1.